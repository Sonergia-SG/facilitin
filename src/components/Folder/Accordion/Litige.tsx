import React from 'react';

interface Props {
  file: { statut: number; id_dp_file: number };
  folderId: number;
  inLitige: any;
  loading: boolean;
}

const Litige = ({
  file, inLitige, folderId, loading,
}: Props) => {
  const isLitige = file.statut === 10;

  return (
    <button
      type="button"
      style={{ transition: 'opacity 0.5s ease' }}
      className={`button is-danger${isLitige ? '' : ' is-outlined'} is-pulled-right${
        loading ? ' is-loading' : ''
      }`}
      id="button-litige"
      disabled={file.statut === 15 || isLitige}
      onClick={() => inLitige(folderId, file.id_dp_file)}
    >
      {'Litige'}
    </button>
  );
};

export default Litige;
