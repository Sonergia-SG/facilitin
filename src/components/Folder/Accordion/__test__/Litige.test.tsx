/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';

import Litige from '../Litige';

describe('Litige button', () => {
  it('render Litige button enabled by default', () => {
    const wrapper = shallow(
      <Litige
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        inLitige={() => 'Hello'}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render Litige button disabled for status 10 (rejected)', () => {
    const wrapper = shallow(
      <Litige
        file={{ statut: 10, id_dp_file: 0 }}
        folderId={0}
        inLitige={() => 'Hello'}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render Litige button disabled for status 15 (accespted)', () => {
    const wrapper = shallow(
      <Litige
        file={{ statut: 15, id_dp_file: 0 }}
        folderId={0}
        inLitige={() => 'Hello'}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render Litige button loading', () => {
    const wrapper = shallow(
      <Litige
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        inLitige={() => 'Hello'}
        loading
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
