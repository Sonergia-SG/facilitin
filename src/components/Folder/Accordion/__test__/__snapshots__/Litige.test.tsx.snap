// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Litige button render Litige button disabled for status 10 (rejected) 1`] = `
<button
  className="button is-danger is-pulled-right"
  disabled={true}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "opacity 0.5s ease",
    }
  }
  type="button"
>
  Litige
</button>
`;

exports[`Litige button render Litige button disabled for status 15 (accespted) 1`] = `
<button
  className="button is-danger is-outlined is-pulled-right"
  disabled={true}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "opacity 0.5s ease",
    }
  }
  type="button"
>
  Litige
</button>
`;

exports[`Litige button render Litige button enabled by default 1`] = `
<button
  className="button is-danger is-outlined is-pulled-right"
  disabled={false}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "opacity 0.5s ease",
    }
  }
  type="button"
>
  Litige
</button>
`;

exports[`Litige button render Litige button loading 1`] = `
<button
  className="button is-danger is-outlined is-pulled-right is-loading"
  disabled={false}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "opacity 0.5s ease",
    }
  }
  type="button"
>
  Litige
</button>
`;
